### DISCUSSION TITLE
why no ignoreUnmappedTargetProperties ?

### DISCUSSION THREAD

**Discussion by EasonLink** [mapstruct/mapstruct#3264] [Q&A]
@Mapper(unmappedTargetPolicy = ReportingPolicy.ERROR, componentModel = MappingConstants.ComponentModel.SPRING)
public interface MapStructMapper {

    List<ListShopWayResp> listPoConvertResp(List<ShopWay> shopWays);

    @Mapping(source = "gmtCreate", target = "createTime")
    GetBasicInfoShopWayResp getBasicInfoPoConvertResp(ShopWay shopWay);

    @Mapping(ignore = true, target = "shopName")
    @BeanMapping(ignoreUnmappedSourceProperties = )// wish ignore target (shopName,userName) but there isn't
    List<ListShopInfoShopWayResp> listShopInfoCommonConvertResp(List<ShopWayUserAmount> users);

}

      I may have made a mistake because  I don't know much about the business scenario. 
      Thank you very much if you can receive your reply ！！！

**Comment by filiphr** [comment]
The reason why there is no `BeanMapping#ignoreUnmappedTargetProperties` is because you can write `@Mapping(target = "shopName", ignore = true)`.

`BeanMapping#ignoreUnmappedSourceProperties` was added because `source` is an optional value in `Mapping` and `target` is a required value in `Mapping`. We didn't want to change this in order to allow `@Mapping(source = "source", ignore = true)`

**Reply by EasonLink** [reply]
Because unmappedTargetPolicy = ReportingPolicy.ERROR.
What do I do if I want the method to ignore the target field and compile without error ?

@mapping(target = "shopName", ignore = true) Can't solve this problem
Because he still compiler error

**Reply by filiphr** [reply]
`@Mapping(target = "shopName", ignore = true)` should work. What kind of a compilation error are you getting?

**Reply by EasonLink** [reply]
```
@Mapper(unmappedTargetPolicy = ReportingPolicy.ERROR, componentModel = MappingConstants.ComponentModel.SPRING)
public interface ShopWayMapStructMapperV2 {

    @Data
    class ListShopInfoShopWayResp {

        private Long shopId;

        private String shopName;

        private String userId;

        private String userName;

        private Integer amount;

    }

    @Data
    class ShopWayUserAmount {

        private Long shopId;

        private String userId;

        private Boolean use;

        private Integer amount;

    }

    @Mapping(target = "shopName", ignore = true)
    @Mapping(target = "userName", ignore = true)
    List<ListShopInfoShopWayResp> listShopInfoCommonConvertResp(List<ShopWayUserAmount> users);

}
```
### error message
**java: Unmapped target properties: "shopName, userName". Mapping from Collection element "ShopWayMapStructMapperV2.ShopWayUserAmount shopWayUserAmount" to "ShopWayMapStructMapperV2.ListShopInfoShopWayResp listShopInfoShopWayResp".**

**Reply by EasonLink** [reply]
I tried but it didn't work, so I assumed that @Mapping(target = "shopName", ignore = true) was invalid

**Reply by filiphr** [reply]
The reason why it is not working is because you are putting it in the wrong place. If you want to customize a mapping of the collection elements then you need to define it.

e.g.

```java
@Mapper(...)
public interface ShopWayMapStructMapperV2 {

    //...


    List<ListShopInfoShopWayResp> listShopInfoCommonConvertResp(List<ShopWayUserAmount> users);

    @Mapping(target = "shopName", ignore = true)
    @Mapping(target = "userName", ignore = true)
    ListShopInfoShopWayResp shopInfoCommonConvertResp(ShopWayUserAmount user);

}
```

**Reply by EasonLink** [reply]
Oh, well, problem solved, because I'm so unskilled, thank you for your answer!

**Reply by meilleureVie** [reply]
there are some cases we have a lot of attributes to ignore and doing like this **@Mapping(target = "shopName", ignore = true)** for each attributes will not help.
Also i'm wondering why **BeanMapping#unmappedTargetPolicy** exists but **BeanMapping#unmappedSourcePolicy** did not exist

**Reply by filiphr** [reply]
If there is a big demand for `BeanMapping#ignoreUnmappedTargetProperties` maybe we should add it, up to now this hasn't been a big priority and we haven't seen a lot of people asking for it.

As for the missing `BeanMapping#unmappedSourcePolicy` I would suggest creating an issue so that we can add it, it has been overlooked.
